File: python/problem_4.py
def largest_palindrome_product(largest_palindrome=0):
    digits_products = (x * y for x in range(900, 1000)
                       for y in range(900, 1000))
    for product in digits_products:
        if product == int(str(product)[::-1]) and product > largest_palindrome:
            largest_palindrome = product
    return largest_palindrome


answer = 906609
print(True if largest_palindrome_product() == answer else False)

File: php/problem_4.php
<?php
function reverse_number($number)
{
  // int -> string -> reverse -> int
  $numberAsString = (string) $number;
  $reverseString = strrev($numberAsString);
  $reverseAsInt = (int) $reverseString;

  return $reverseAsInt;
}

function largest_palindrome_product($largest_palindrome = 0)
{
  $digits_product = [];
  $loop_range = range(900, 999);

  // generate products with 3-digit numbers
  foreach ($loop_range as $num_one) {
    foreach ($loop_range as $num_two) {
      $digits_product[] = $num_one * $num_two;
    }
  }

  // checks every generated product
  foreach ($digits_product as $number) {
    $reversed_number = reverse_number($number);

    if ($number == $reversed_number and $number > $largest_palindrome) {
      $largest_palindrome = $number;
    }
  }

  return $largest_palindrome;
}

$anwser = 906609;
echo largest_palindrome_product() == $anwser ? "true" : "false";
